Server/src/models/DataUploadModel.ts

```diff
@@ -15,190 +15,190 @@ import { Material } from './entities/Material';
-    await connection.manager.save(book);
-    return book.id;
-}
+    async insertReferenceType(preferenceType: string): Promise<number> {
 
-const insertPublisher = async (publisher) => {
+        const connection = getConnection();
 
-    const connection = getConnection();
+        let book = new Publicationtype();
+        book.id;
+        book.name = preferenceType;
+        await connection.manager.save(book);
+        return book.id;
+    }
 
-    let publisherName = new Publisher();
-    publisherName.id;
-    publisherName.name = publisher;
-    await connection.manager.save(publisherName);
-    return publisherName.id;
-}
+    async insertPublisher(publisher: string): Promise<number> {
+
+        const connection = getConnection();
+
+        let publisherName = new Publisher();
+        publisherName.id;
+        publisherName.name = publisher;
+        await connection.manager.save(publisherName);
+        return publisherName.id;
+    }
 
-const insertAuthors = async (storeAuthors) => {
+    async insertAuthors(storeAuthors: IAuthors[]) {
 
-    const connection = getConnection();
+        const connection = getConnection();
 
-    for (var i = 0; i < storeAuthors.length; i++) {
-        console.log(storeAuthors[i].firstname);
-        let author = new Authors();
-        author.id;
-        author.firstName = storeAuthors[i].firstname;
-        author.lastName = storeAuthors[i].lastname;
-        author.middleName = storeAuthors[i].middlename;
-        await connection.manager.save(author);
+        console.log(storeAuthors);
+
+        for (let i = 0; i < storeAuthors.length; i++) {
+            console.log(storeAuthors[i].firstname);
+            let author = new Authors();
+            author.id;
+            author.firstName = storeAuthors[i].firstname;
+            author.lastName = storeAuthors[i].lastname;
+            author.middleName = storeAuthors[i].middlename;
+            await connection.manager.save(author);
+        }
     }
-}
 
-const insertPublication = async (referenceTitle, referencePages, preferenceTypeID, publisherNameId, referenceYear, referenceVolume, referenceAuthors) => {
-
-    const connection = getConnection();
-
-    let publication = new Publications();
-    publication.id;
-    publication.name = referenceTitle;
-    publication.doi;
-    publication.pages = referencePages;
-    publication.publicationtypeId = preferenceTypeID;
-    publication.publisherId = publisherNameId;
-    publication.year = referenceYear;
-    publication.volume = referenceVolume;
-    publication.datePublished;
-    publication.dateAccessed;
-    publication.authors = referenceAuthors;
-    await connection.manager.save(publication);
-    console.log("publication saved.");
-    return publication.id;
-}
+    async insertPublication(referenceTitle: string, referencePages: number, preferenceTypeID: number, publisherNameId: number, referenceYear: number, referenceVolume: number, referenceAuthors: any[]): Promise<number> {
```

Comment:
cant remember what we decided about this, just reply here

Reply:
im talking referenceAuthors: any[]

Reply:
The entity requires additional fields to assign to a property of the entity ie: requires date for example, so the type check fails